### PR TITLE
Release/2024 02 23

### PR DESCRIPTION
**Pull request by ludtkemorgan**
# Pull Request Template

## Issue Overview

This PR addresses [#389](https://github.com/bloom-housing/bloom/issues/3893)

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR pulls over the prisma changes from core and also has the majority of the changes specific to Doorway.

*Note* working through the following still so the PR isn't 100% ready, but can be looked at by the team to find any issues.

What is still remaining:
- ~Filtering on the listings page~
- ~Tests working in github actions~
- ~Updating the docker images for backend~
- Pull in the mandated account changes

## How Can This Be Tested/Reviewed?

This can only be tested locally and all flows on both public/partner sites should be tested (excluding the items in "what is still remaining" above)

Notes to test this locally:
- Make sure you have a `.env` file in api directory with the values from `.env.template`. You can use the same values you might have setup in core. But note there are a few values that are specific to Doorway such as the values to publish to s3. If you don't already have those values reach out to Morgan
- The best way to get setup is to do the following:
  - run `yarn install` from api directory
  - run `yarn prisma generate` from api directory
  - run `yarn db:reseed:with-external:dev` from api directory
  - run `yarn dev:all` from root directory (make sure you have also installed at root)

Specific areas that should be tested extra closely:
- Adding images to a listing (make sure that the aws flow works)
- Filtering on the public listings page. This was rewritten so making sure that no functionality on the listing page is broken

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by emilyjablonski** [comment]
`yarn setup:dev` doesn't work for me here, are you seeing something different?
Running seed command `ts-node prisma/seed.ts --environment development --jurisdictionName Bay Area` ...
TypeError [ERR_PARSE_ARGS_UNEXPECTED_POSITIONAL]: Unexpected argument 'Area'. This command does not take positional arguments

**Comment by emilyjablonski** [comment]
@ludtkemorgan That worked thank you! It could just be me, but as I'm poking around I'm noticing that the partners site is moving very slowly - it could be something unrelated, but just curious if others notice the same / wanted to flag

**Comment by ColinBuyck** [comment]
Screenshot 2024-03-08 at 10 33 41 AM
Can't seem to get the requested changes email to send

**Comment by ColinBuyck** [comment]
For documentation sake, I found that the listing image upload was erroring if the name was greater than or equal to 128 characters

**Comment by ludtkemorgan** [comment]
> Screenshot 2024-03-08 at 10 33 41 AM Can't seem to get the requested changes email to send

It was determined that this is also a bug in core. A PR has been made there to address it https://github.com/bloom-housing/bloom/pull/3934

File: api/src/services/listing.service.ts [hunk 1]
```diff
@@ -229,6 +230,122 @@ export class ListingService implements OnModuleInit {
...
+      currentPage: number;
+      itemCount: number;
+      itemsPerPage: number;
+      totalItems: number;
+      totalPages: number;
+    };
+  }> {
+    const onlyLettersPattern = /^[A-Za-z ]+$/;
+    const whereClauseArray = [];
+    if (params?.filter?.length) {
```
Comment (YazeedLoonat) [inline api/src/services/listing.service.ts:245]: looks good!

**Review by YazeedLoonat — CHANGES_REQUESTED** [review]
soem minor questions looks good overall!

File: api/src/services/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,852 @@
...
+        mapTo(User, storedUser),
+        dto.appUrl,
+        confirmationUrl,
+        dto.newEmail,
+      );
+    }
+
+    // only update userRoles if something has changed
+    if (dto.userRoles && storedUser.userRoles) {
+      if (
```
Comment (ColinBuyck) [inline api/src/services/user.service.ts:199]: Think we should include the recent security changes in this release

File: api/Procfile [hunk 3]
```diff
@@ -0,0 +1 @@
+web: yarn start:prod
```
[the remark this answers is not in the document]
Comment (ludtkemorgan) [reply api/Procfile:1]: This isn't needed, but I didn't bother deleting any of the files for the core setup such as Heroku and circleCI files. That is similar to what we did with the old backend

**Comment by ludtkemorgan** [comment]
@YazeedLoonat Your comments have been addressed. I also added a few more things since your last review:
- Pulled in the latest commits to main.
- Added the listing opportunity flow (govDelivery)
- Realized there was one more field needed on the combined listings table (reserved community types)

**Review by YazeedLoonat — APPROVED** [review]
LGTM

**Pull request merged**